linkerd/core/src/main/scala/io/buoyant/linkerd/RequestEvaluator.scala

```diff
@@ -0,0 +1,168 @@
+package io.buoyant.linkerd
+
+import com.twitter.finagle._
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.http._
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.util._
+import io.buoyant.namer.DelegateTree._
+import io.buoyant.namer.{DelegateTree, Delegator}
+import io.buoyant.router.RouterLabel
+import io.buoyant.router.RoutingFactory
+import io.buoyant.router.RoutingFactory.BaseDtab
+import io.buoyant.router.context.{DstBoundCtx, DstPathCtx}
+
+class RequestEvaluator(
+  endpoint: EndpointAddr,
+  namer: DstBindingFactory.Namer,
+  dtab: BaseDtab,
+  label: String
+) extends SimpleFilter[Request, Response] {
+
+  private[this] val RequestTracerMaxDepthHeader = "l5d-max-depth"
+
+  private[this] def printEvaluatedRequest(
+    prependText: String,
+    routerLabel: String,
```

Comment:
I think we can omit this here and access the classes constructor parameter directly.